index.js

```diff
@@ -37,7 +39,9 @@ function compose (middleware) {
     function dispatch (i) {
       if (i <= index) return Promise.reject(new Error('next() called multiple times'))
       index = i
-      const fn = middleware[i] || next
+      let fn
+      if (wrapper) fn = middleware[i] ? wrapper(middleware[i]) : next
```

Comment:
`next` is guaranteed to be a function in conformance with standard Koa practice. It's not coming from the user 90% of the time. If you use `co.wrap` as a wrapper, you don't want it processing next.